builtin/providers/aws/resource_aws_emr.go

```diff
@@ -0,0 +1,413 @@
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+					},
+				},
+			},
+			"tags": tagsSchema(),
+			"configurations": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"service_role": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsEMRCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).emrconn
+
+	log.Printf("[DEBUG] Creating EMR cluster")
+	masterInstanceType := d.Get("master_instance_type").(string)
+	coreInstanceType := masterInstanceType
+	if v, ok := d.GetOk("core_instance_type"); ok {
+		coreInstanceType = v.(string)
+	}
+	coreInstanceCount := d.Get("core_instance_count").(int)
+
+	applications := d.Get("applications").(*schema.Set).List()
+	var userKey, subnet, extraMasterSecGrp, extraSlaveSecGrp, emrMasterSecGrp, emrSlaveSecGrp, instanceProfile, serviceRole string
+	instanceProfile = "EMR_EC2_DefaultRole"
+	ec2Attributes := d.Get("ec2_attributes").([]interface{})
+	if len(ec2Attributes) == 1 {
+		attributes := ec2Attributes[0].(map[string]interface{})
+		userKey = attributes["key_name"].(string)
+		subnet = attributes["subnet_id"].(string)
+		extraMasterSecGrp = attributes["additional_master_security_groups"].(string)
+		extraSlaveSecGrp = attributes["additional_slave_security_groups"].(string)
+		emrMasterSecGrp = attributes["emr_managed_master_security_group"].(string)
+		emrSlaveSecGrp = attributes["emr_managed_slave_security_group"].(string)
+
+		if len(strings.TrimSpace(attributes["instance_profile"].(string))) != 0 {
+			instanceProfile = strings.TrimSpace(attributes["instance_profile"].(string))
+		}
+	}
+
+	if v, ok := d.GetOk("service_role"); ok {
+		serviceRole = v.(string)
+	} else {
+		serviceRole = "EMR_DefaultRole"
+	}
+
+	emrApps := expandApplications(applications)
+
+	params := &emr.RunJobFlowInput{
+		Instances: &emr.JobFlowInstancesConfig{
+			Ec2KeyName:                  aws.String(userKey),
+			Ec2SubnetId:                 aws.String(subnet),
+			InstanceCount:               aws.Int64(int64(coreInstanceCount + 1)),
+			KeepJobFlowAliveWhenNoSteps: aws.Bool(true),
+			MasterInstanceType:          aws.String(masterInstanceType),
+			SlaveInstanceType:           aws.String(coreInstanceType),
+			TerminationProtected:        aws.Bool(false),
+			AdditionalMasterSecurityGroups: []*string{
+				aws.String(extraMasterSecGrp),
+			},
+			AdditionalSlaveSecurityGroups: []*string{
+				aws.String(extraSlaveSecGrp),
+			},
+			EmrManagedMasterSecurityGroup: aws.String(emrMasterSecGrp),
+			EmrManagedSlaveSecurityGroup:  aws.String(emrSlaveSecGrp),
+		},
+		Name:         aws.String(d.Get("name").(string)),
+		Applications: emrApps,
+
+		JobFlowRole:       aws.String(instanceProfile),
+		ReleaseLabel:      aws.String(d.Get("release_label").(string)),
+		ServiceRole:       aws.String(serviceRole),
+		VisibleToAllUsers: aws.Bool(true),
```

Comment:
This seems like an optional attribute we may want to expose, but with a default of `true`. Not a blocker thought 😄